### PR TITLE
Spec outline for Ethereum-proxy

### PR DESCRIPTION
**Pull request by gadcl**

### CODE REVIEW

File: behaviors/services/crosschain-connector.md [hunk 1]
```diff
@@ -42,3 +42,47 @@ Runs nodes for other blockchains like Ethereum and provides read access to them.
...
+* Returns the block info (uint64 block_number, uint64 block_timestamp) of the resulting block.
+
+
+
+&nbsp;
+## `EthereumGetLogs` (method)
+> Returns all Ethereum logs entries which comply to the given filter params. 
+* Query the Ethereum node to find the appropriate logs using the given arguments.
+* Comply to the finality time shift:
+    * Get Latest finalized block by calling EthereumGetLatestBlockInfo
```
Comment (OdedWx) [inline behaviors/services/crosschain-connector.md]: No actual call to EthereumGetLatestBlockInfo, the logic is in the connector

File: behaviors/services/crosschain-connector.md [hunk 2]
```diff
@@ -42,3 +42,47 @@ Runs nodes for other blockchains like Ethereum and provides read access to them.
...
+    * Verify the block range provided is in the finality range.
+        * finalized block number >  until_block_number (provided in arguments).
```
Comment (OdedWx) [inline behaviors/services/crosschain-connector.md]: until_block_number = min(finalized block number, until_block_number)

File: behaviors/services/crosschain-connector.md [hunk 3]
```diff
@@ -42,3 +42,47 @@ Runs nodes for other blockchains like Ethereum and provides read access to them.
...
+* Derive the event signature key from the provided contract ABI using the event_name.
+* The ABI needs to be defined for the ethereum connector to be able to make the call
+  * The ABI must contain only one event with the provided event name.
+* Returns:
+    * call_result - indicates the call success or failure.
+    * block_number - indicates the last filtered block_number.
```
Comment (OdedWx) [inline behaviors/services/crosschain-connector.md]: block_number - indicates the last filtered block_number. The output block number is less or equal to the requested until_block

File: behaviors/v1_1/ethereum-proxy.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+* Retrieve the result by calling `EthereumConnector.EthereumGetBlockInfoByTime(block.timestamp, ethereum_timestamp)`.
+* On error return nil.
+
+#### ether.GetLatestBlockInfo() : (uint64 block_number, uint64 block_timestamp).
+> Returns the block info (number, timestamp) of the latest Ethereum block which complies with the finality time shift (reference_time := block.timestamp - ETHEREUM_FINALITY_TIME_COMPONENT).
+* Output: Ethereum block number and its corresponding timestamp.
+* Retrieve the result by calling `EthereumConnector.EthereumGetLatestBlockInfo(block.timestamp)`.
+* On error return nil.
+
+#### ether.GetPastEvents(contract_address, event_name, contract_ABI, from_block, until_block) : (log_record_list, block_number, call_result)
```
Comment (OdedWx) [inline behaviors/v1_1/ethereum-proxy.md]: ethereum.getLogs / filterLogs

File: behaviors/smart-contracts/system/_ElectionsProxy.md [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+
+#### PoS state queries 
+* getDelegatorInfo(address) : delegatorInfo 
+    * delegator_info := (address, balance, stake, cool_down, delegate_to) 
+* getGuardians() : list of addresses
+* getGuardianInfo(address) : guardianInfo
+    * guardian_info := (address, vote)
+        * vote := (timestamp, list of addresses (to vote out))
+* getValidators() : list of addresses
+* getValidatorInfo(address) : validatorInfo
```
Comment (noambergIL) [inline behaviors/smart-contracts/system/_ElectionsProxy.md]: validators don't have a vote ... if then vote they will be in the delegators.

**Comment by gadcl** [comment]
irrelevant

**Pull request closed without merging**